### PR TITLE
feat(e2e): add init e2e tests

### PR DESCRIPTION
**Pull request by spencercorwin**
### Description of the Change

Adds E2E tests for `neo-one init` command. Basically just pulls everything from #1705, which was created pre-RushJS, and makes it work with the current NEO•ONE and RushJS setup.

### Test Plan

`rush e2e -t packages/neo-one-cli/src/__e2e__/cmd/init.test.ts`

### Alternate Designs

None.

### Benefits

E2E test for init command.

### Possible Drawbacks

None.

### Applicable Issues

#1682 
#1705

### CODE REVIEW

File: common/changes/@neo-one/cli/init-e2e_2020-06-16-23-17.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@neo-one/cli",
+      "comment": "Add E2E test for neo-one init command.",
+      "type": "patch"
```
Comment (danwbyrne) [inline common/changes/@neo-one/cli/init-e2e_2020-06-16-23-17.json:6]: adding a test should just be a `none` type I think

File: packages/neo-one-build-tools/types/e2e.d.ts [hunk 2]
```diff
@@ -3,14 +3,32 @@ declare interface NodeProject {
...
+  readonly exec: (command: string, options?: object) => Promise<void>;
+  readonly env: any;
+}
+
 declare interface One {
   readonly addCleanup: (callback: () => Promise<void> | void) => void;
   readonly cleanupTest: () => Promise<void>;
+  readonly setup: () => Promise<void>;
+  readonly setupPorts: () => Promise<void>;
+  readonly teardown: () => Promise<void>;
```
Comment (danwbyrne) [inline packages/neo-one-build-tools/types/e2e.d.ts:16]: whats up with this

File: packages/neo-one-build-tools/types/e2e.d.ts [hunk 3]
```diff
@@ -3,14 +3,32 @@ declare interface NodeProject {
...
   readonly createExec: (project: string) => (command: string, options?: object) => Promise<string>;
+  readonly execBase: (
+    commandIn: string,
+    project: string,
+    options?: object,
+  ) => import('execa').ExecaChildProcess<string>;
+  readonly execBaseNode: (commandIn: string, environment: object) => import('execa').ExecaChildProcess<string>;
```
Comment (danwbyrne) [inline packages/neo-one-build-tools/types/e2e.d.ts:23]: ?

File: packages/neo-one-build-tools/types/e2e.d.ts [hunk 4]
```diff
@@ -3,14 +3,32 @@ declare interface NodeProject {
...
   readonly createExecAsync: (project: string) => (command: string, options?: object) => void;
   readonly createNodeProject: (project: string) => NodeProject;
+  readonly createCLIProject: (project: string) => CLIProject;
+  readonly getTempDir: (project: string) => string;
+  readonly getAvailableMinPortForRange: (range: number, start?: number) => number;
```
Comment (danwbyrne) [inline packages/neo-one-build-tools/types/e2e.d.ts:28]: ?

File: packages/neo-one-build-tools/types/e2e.d.ts [hunk 5]
```diff
@@ -3,14 +3,32 @@ declare interface NodeProject {
...
   readonly until: (func: () => Promise<void>) => Promise<void>;
   readonly measureRequire: (mod: string) => Promise<number>;
   readonly getProjectConfig: (project: string) => any;
+  readonly getProjectDir: (project: string) => string;
```
Comment (danwbyrne) [inline packages/neo-one-build-tools/types/e2e.d.ts:32]: ?
